Standardize on manual instrumentation intro text (#2852)
open-telemetry/opentelemetry.io: content/en/docs/instrumentation/cpp/manual.md

title: Manual Instrumentation
linkTitle: Manual
weight: 30
description: Manual instrumentation for OpenTelemetry C++
---

> [Manual instrumentation](https://opentelemetry.io/docs/concepts/instrumentation/manual/)

Reviewer: ```suggestion
[Manual instrumentation](/docs/concepts/instrumentation/manual/)
```

> is the process of adding observability code to your application yourself. If you
> are developing a standalone process or service, then you would take a dependency
> on the API and the SDK.

Reviewer: Start a new paragraph:

```suggestion
is the process of adding observability code to your application yourself.

If you
are developing a standalone process or service, then you would take a dependency
on the API and the SDK.
```

Also, I can't quite make sense of the wording of the last sentence.

Author: I reordered the original sentence, because most people come because the develop a standalone service, although that makes things harder to understand. Let me think about that for a little bit longer (or if anyone has good suggestions on the wording, please do :))

Author: I rewrote the whole text, outdating this discussion, will mark it as resolved

> If you are developing a library, that wants to export telemetry data using

Reviewer: ```suggestion
If you are developing a library and want it to export telemetry data using
```

> OpenTelemetry, it MUST only depend on the API and should never configure or

Reviewer: ```suggestion
OpenTelemetry, it **must** depend only on the API and should never configure or
```

> depend on the OpenTelemetry SDK.
> 
> This way libraries will obtain a real implementation of OpenTelemetry only if
> the user application is configured for it and otherwise will function as without
> added observability.

Reviewer: Combine the paragraphs since they are both dealing with libraries?

```suggestion
depend on the OpenTelemetry SDK. In this way, libraries will obtain a real implementation of OpenTelemetry only if
the user application is configured for it and otherwise will function as without
added observability.
```

I can't quite make sense of this last sentence either.

Author: > In this way, libraries will obtain a real implementation of OpenTelemetry only if the user application is configured for it

This was the original sentence I copied from one of the language SIG pages. I think the message is that the library making only use of the API works just as if the API was never used until someone adds the SDK to the application. It's really hard to express that in a short sentence. But we can drop it, since it is explained in more detail in the library guides.

> For more details, check out the
> [Library Guidelines](https://opentelemetry.io/docs/concepts/instrumentation/libraries/).

Reviewer: ```suggestion
For more details, see
[Library Guidelines](/docs/concepts/instrumentation/libraries/).
```
